galgebra/ga.py

```diff
@@ -93,9 +93,6 @@ def update_and_substitute(expr1, expr2, mul_dict):
     product of ``bases1[i]*bases2[j]`` as a linear combination of scalars and
     bases of the geometric algebra.
     """
-    if (isinstance(expr1, numbers.Number) or expr1.is_commutative) \
-        or (isinstance(expr2, numbers.Number) or expr2.is_commutative):
-        return expr1 * expr2
```

Comment:
This part was actually incorrect for for when `mul_dict[S(1), S(1)] != S(1)`, like it is for the hestenes dot

Reply:
I suppose strictly this change could go in all by itself